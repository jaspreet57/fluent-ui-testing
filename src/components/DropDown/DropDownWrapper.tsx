import React from 'react'
import { Dropdown, IDropdownOption } from 'office-ui-fabric-react';

interface DropDownWrapperProps {
    handleChange: (event: React.FormEvent<HTMLDivElement>, option?: IDropdownOption | undefined, index?: number | undefined) => void;
    options: IDropdownOption[]
    [key: string]: any
}

const DropDownWrapper: React.FC<DropDownWrapperProps> = ({ handleChange, options, ...props }) => {
    return (
        <Dropdown
            placeholder="Required with no label"
            ariaLabel="Required dropdown example"
            onChange={handleChange}
            options={options}
            required={true}
            styles={{ dropdown: { width: 300 } }}
            {...props}
        />
    )
}

export {
    DropDownWrapper
}