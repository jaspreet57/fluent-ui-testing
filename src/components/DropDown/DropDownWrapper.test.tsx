import React from "react";
import { DropDownWrapper } from "./DropDownWrapper";
import toJson from "enzyme-to-json";
import { shallow, mount } from "enzyme";

const defaultProps = [
  { key: "A", text: "Option a", title: "I am option a." },
  { key: "B", text: "Option b" },
  { key: "C", text: "Option c", disabled: true },
  { key: "D", text: "Option d" },
  { key: "E", text: "Option e" },
];

describe("DropDownWrapper", () => {
  it("should match snapshots", () => {
    const handleFn = jest.fn();
    const wrapper = shallow(
      <DropDownWrapper handleChange={handleFn} options={defaultProps} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should call handleChange with selected prop", () => {
    const handleFn = jest.fn();
    const wrapper = mount(
      <DropDownWrapper handleChange={handleFn} options={defaultProps} />
    );

    wrapper.find('.ms-Dropdown-title').simulate('click', { nativeEvent: { stopImmediatePropagation: jest.fn() } })
    wrapper.find('.ms-Dropdown-item').at(0).simulate('click', { nativeEvent: { stopImmediatePropagation: jest.fn() } })

    expect(handleFn).toHaveBeenCalledTimes(1)
    expect(handleFn).toHaveBeenCalledWith(expect.anything(), defaultProps[0], 0)
  });
});
