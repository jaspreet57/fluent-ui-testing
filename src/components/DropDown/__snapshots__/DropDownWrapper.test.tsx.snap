// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownWrapper should match snapshots 1`] = `
<StyledWithResponsiveMode
  ariaLabel="Required dropdown example"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "key": "A",
        "text": "Option a",
        "title": "I am option a.",
      },
      Object {
        "key": "B",
        "text": "Option b",
      },
      Object {
        "disabled": true,
        "key": "C",
        "text": "Option c",
      },
      Object {
        "key": "D",
        "text": "Option d",
      },
      Object {
        "key": "E",
        "text": "Option e",
      },
    ]
  }
  placeholder="Required with no label"
  required={true}
  styles={
    Object {
      "dropdown": Object {
        "width": 300,
      },
    }
  }
/>
`;
